### PR TITLE
AB#65910 AHP Housing Type unique name validation + merge Common UI projects

### PR DESCRIPTION
**Pull request by jjankowski87**
image

### CODE REVIEW

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Contract/HomeTypes/Queries/GetHomeTypeQuery.cs [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 namespace HE.Investment.AHP.Contract.HomeTypes.Queries;
 
-public record GetHomeTypeQuery(string FinancialSchemeId, string HomeTypeId) : IRequest<HomeType>;
+public record GetHomeTypeQuery(string SchemeId, string HomeTypeId) : IRequest<HomeType>;
```
Comment (Carq) [inline src/HE.Investment.AHP.Contract/HomeTypes/Queries/GetHomeTypeQuery.cs:5]: Maybe we could use ValueObject For SchemeId and for HomeTypeId? They will be often used

File: src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/HomeTypeCommandHandlerBase.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private readonly ILogger _logger;
+
+    protected HomeTypeCommandHandlerBase(ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    protected IList<ErrorItem> PerformWithValidation(params Action[] actions)
+    {
+        var errors = new List<ErrorItem>();
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/HomeTypeCommandHandlerBase.cs:18]: errors is not used?

File: src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/HomeTypeCommandHandlerBase.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        foreach (var action in actions)
+        {
+            try
+            {
+                action();
+            }
+            catch (DomainValidationException domainValidationException)
+            {
+                _logger.LogWarning(domainValidationException, "Validation error(s) occured: {Message}", domainValidationException.Message);
+                return domainValidationException.OperationResult.Errors;
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/HomeTypeCommandHandlerBase.cs:28]: Probably here you should add errors to `errors` variable?

File: src/HE.Investments.Common.WWW/HE.Investments.Common.WWW.csproj [hunk 4]
```diff
@@ -13,6 +13,7 @@
 
   <PropertyGroup>
     <NoWarn>1591</NoWarn>
+    <NoWarn>SA1118</NoWarn>
```
Comment (Carq) [inline src/HE.Investments.Common.WWW/HE.Investments.Common.WWW.csproj:16]: It should be added to .editorconfig

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.Contract/HomeTypes/Queries/GetHomeTypeQuery.cs (hunk 1 above)
Comment (jjankowski87) [reply src/HE.Investment.AHP.Contract/HomeTypes/Queries/GetHomeTypeQuery.cs:5]: This is query in contract, SchemeId and HomeTypeId is ValueObject in domain. Contract cannot use Domain

File: src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/HomeTypeCommandHandlerBase.cs (hunk 2 above)
Comment (jjankowski87) [reply src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/HomeTypeCommandHandlerBase.cs:18]: mistake, will be fixed

**Pull request merged**